com.avaloq.tools.ddk.xtext/src/com/avaloq/tools/ddk/xtext/scoping/DelegatingScope.java

```diff
@@ -32,6 +35,101 @@
 public class DelegatingScope extends AbstractRecursiveScope {
 
   private static final Logger LOGGER = Logger.getLogger(DelegatingScope.class);
+
+  /**
+   * Stores the scope ID to be logged by {@link #STACK_OVERFLOW_EOBJECT_DESCRIPTION}.
+   * <p>
+   * If multiple threads were to concurrently run into {@link StackOverflowError} it is possible that the threads would log the wrong scope IDs. But this is
+   * very unlikely to happen and the scope ID is only logged to assist the developer in addressing the programming error.
+   */
+  private static String stackOverflowScopeId;

```

Comment:
Maybe we could get back to this static field again at some point. Maybe ThreadLocal<String> instead of String? Can discuss for a followup.